test/tests/document.test.js

```diff
@@ -0,0 +1,48 @@
+import { expect } from 'chai'
+import request from 'request'
+import { bootServer, mockApi } from '../utils'
+import pageData from '../mocks/page.json'
+import style from '../../dist/default-style'
+
+// test Tapestry components and data
+describe('HTML document', () => {
+
+  let tapestry = null
+  let config = {
+    siteUrl: 'http://dummy.api:80'
+  }
+
+
+  beforeEach(done => {
+    mockApi({
+      path: '/wp-json/wp/v2/pages',
+      query: { filter: { name: 'home' }}
+    })
+    tapestry = bootServer(config, done)
+  })
+  afterEach(() => tapestry.stop())
+
+
+  it('Data is correctly loaded on page', (done) => {
+    console.log(tapestry.server.info.uri)
+    request
+      .get(tapestry.server.info.uri, (err, res, body) => {
+        expect(body).to.contain(`window.__ASYNC_PROPS__ = [{"resp":${JSON.stringify(pageData)}}]`)
```

Comment:
Never occurred to me how awesome isomorphism/universalism can be for testing as well.